Lib/test/test_mmap.py

```diff
@@ -255,10 +255,15 @@ def test_access_parameter(self):
             # Try writing with PROT_EXEC and without PROT_WRITE
             prot = mmap.PROT_READ | getattr(mmap, 'PROT_EXEC', 0)
             with open(TESTFN, "r+b") as f:
-                m = mmap.mmap(f.fileno(), mapsize, prot=prot)
-                self.assertRaises(TypeError, m.write, b"abcdef")
-                self.assertRaises(TypeError, m.write_byte, 0)
-                m.close()
+                try:
+                    m = mmap.mmap(f.fileno(), mapsize, prot=prot)
+                except PermissionError:
+                    # on macOS 14, PROT_READ | PROT_WRITE is not allowed
```

Comment:
The test is explicitly *not* setting PROT_WRITE (see line 255), so this comment seems incorrect. It's also catching and ignoring this exception on all posix platforms, not just macOS 14 or even just macOS. Does it really make sense to ignore it that way?

Reply:
Crap, it's a typo: you should read PROT_READ | PROT_EXEC.

Reply:
> Does it really make sense to ignore it that way?

Python cannot argue with kernel security changes towards blocking more attacks. The trend is to block more and more sensitive code patterns like memory pages which can be used for something other than just executing them. If you can READ + EXEC, ROP gadgets are easier to discover for example.

Reply:
If the test is changed, it should be changed first in the main branch.

The change got approved in the main branch. IMO it's ok to skip the test if mmap() fails with PermissionError. The Python test suite should not become a test of the operating system itself, but only of the small glue on top of the OS. If the OS decides that PROT_READ | PROT_EXEC raises a PermissionError, we can just skip the test.

Well, that's my opinion :-) I just wanted to fix macOS 14 support.